src/linter/linter.ts

```diff
@@ -273,3 +280,90 @@ async function fileExists(dirPath: string) {
 function sortLintResults(lintResults: LintResult[]) {
 	lintResults.sort((a, b) => a.filePath.localeCompare(b.filePath));
 }
+
+/**
+ * Function to check if a file path matches the patterns
+*/
+function isFileIncluded(file: string, patterns: string[]) {
+	let include = true;
+
+	for (const pattern of patterns) {
+		if (pattern.startsWith("!")) {
+			// Handle negation: exclude if the file matches the pattern
+			if (minimatch(file, pattern.slice(1))) {
```

Comment:
Since the patterns stay the same same for all files, I think it make sense to create one minimatch instance per pattern upfront and have them in an array to go through here

Especially for larger patterns this can save us some time. You can tell negative patterns apart from the others using the [`negate`](https://github.com/isaacs/minimatch?tab=readme-ov-file#properties) property. And use the option [`{nonegate: true}`](https://github.com/isaacs/minimatch?tab=readme-ov-file#nonegate) to still have the behavior where it returns true if the (negated) pattern matches as currently archived using the `slice(1)` call

Reply:
Thanks!
Now it's been optimized. Actually, the flipNegate flag did the work